### PR TITLE
[feat]: Support client.getPartitionsForTopic (#320)

### PR DESCRIPTION
**Pull request by XXXMrG**
Master Issue: #320 

### Motivation

Support api: [client.getPartitionsForTopic](https://pulsar.apache.org/api/cpp/3.1.x/classpulsar_1_1_client.html#a208629aff52395ad0072622ae82f5657)

### Modifications

Just bind CPP function: `pulsar_client_get_topic_partitions_async` to JS.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

This change added tests and can be verified as follows:

  - [x] will add end to end test later.

### Documentation

- [x] `doc-required` 
(Your PR needs to update docs and you will update later)

- [ ] `doc-not-needed` 
(Please explain why)

- [ ] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### CODE REVIEW

**Comment by shibd** [comment]
@XXXMrG Thanks for your contribution. Let's wait for your `TODO` list to be done.

**Comment by XXXMrG** [comment]
Done. Would you mind giving my code a quick review? @shibd

File: tests/end_to_end.test.js [hunk 1]
```diff
@@ -1242,5 +1242,21 @@ const Pulsar = require('../index.js');
       await consumer.close();
       await client.close();
     });
+
+    test('Client/getPartitionsForTopic', async () => {
```
Comment (shibd) [inline tests/end_to_end.test.js:1246]: This test should belong `client.test.js`.

File: tests/end_to_end.test.js [hunk 2]
```diff
@@ -1242,5 +1242,21 @@ const Pulsar = require('../index.js');
...
+      const client = new Pulsar.Client({
+        serviceUrl: 'pulsar://localhost:6650',
+        operationTimeoutSeconds: 30,
+      });
+
+      const topic = 'persistent://public/default/get-partitions-for-topic';
+      await client.createProducer({
```
Comment (shibd) [inline tests/end_to_end.test.js:1253]: You can use admin [REST API](https://pulsar.apache.org/admin-rest-api/?version=3.0.0#) to create specified partitions topic, and then to assert `getPartitionsForTopic`.

**Review by shibd — COMMENTED** [review]
Thanks! Left some small comments.

**Comment by XXXMrG** [comment]
@shibd 
I have modified the code according to your comments.
Also, I noticed the compile failed on Windows x86 platforms. I will check it later.

**Comment by shibd** [comment]
> @shibd I have modified the code according to your comments. Also, I noticed the compile failed on Windows x86 platforms. I will check it later.

Nice, Let's run ci look at it.

**Comment by XXXMrG** [comment]
@shibd 
I have fixed the compile problem in Windows x86 platforms.

But I found that the `Client/getPartitionsForTopic` test will fail in the CI environment due to an unexpected extra item that duplicates the topic name in the returned result, it does not happened in my device, this may be related to the Pulsar version in the CI environment, which is `2.10.2`, and i use pulsar `2.11.x`.
I need some assistance to resolve this issue, thx.

**Comment by shibd** [comment]
```
2023-05-09T02:41:35,995+0000 [pulsar-web-48-8] WARN  org.eclipse.jetty.server.HttpChannelState - unhandled due to prior sendError
javax.servlet.ServletException: javax.servlet.ServletException: java.lang.NumberFormatException: For input string: ""4""
	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:162) ~[org.eclipse.jetty-jetty-server-9.4.48.v20220622.jar:9.4.48.v20220622]
	at org.eclipse.jetty.server.handler.StatisticsHandler.handle(StatisticsHandler.java:181) ~[org.eclipse.jetty-jetty-server-9.4.48.v20220622.jar:9.4.48.v20220622]
```

Maybe the request body format is invalid. Will cause the partition topic to create failed.

1. You should assert the HTTP response code.
2. You should be change request body format:

```diff
-          data: '4',
+          data: 4,
```

File: tests/client.test.js [hunk 3]
```diff
@@ -49,5 +74,54 @@ const Pulsar = require('../index.js');
...
+
+        // test on nonPartitionedTopic
+        const nonPartitionedTopicName = 'test-non-partitioned-topic';
+        const nonPartitionedTopic = `persistent://public/default/${nonPartitionedTopicName}`;
+        await requestAdminApi(`${baseUrl}/admin/v2/persistent/public/default/${nonPartitionedTopicName}`, {
+          headers: {
+            'Content-Type': 'application/json',
+          },
+        });
+        const nonPartitionedTopicList = await client.getPartitionsForTopic(nonPartitionedTopic);
```
Comment (shibd) [inline tests/client.test.js:103]: Before this, you should assert the HTTP response code first.

Comment (XXXMrG) [reply tests/client.test.js:103]: Got it, let me try again.

**Comment by XXXMrG** [comment]
@shibd Done. let's run ci check again.

**Comment by XXXMrG** [comment]
@shibd 
It looks like all checks have passed. Can we merge this pull request now?

**Review by shibd — APPROVED** [review]
